gocourse04/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	Ungulates        Types   = "Ungulates"
+	Birds            Types   = "Birds"
+	Primates         Types   = "Primates"
+	GroomingBrush    Purpose = "Grooming Brush"
+	WaterBucket      Purpose = "Water Bucket"
+	VeterinarySyring Purpose = "Veterinary Syring"
+)
+
+type Zoo struct {
+	Areas Areas
+}
+
+func (z Zoo) FindAnimalByName(name string) {
+finish:
+	for _, area := range z.Areas {
+		for sectorName, sector := range area.Sectors {
+			for _, animal := range sector.Animals {
+				if animal.Name == name {
+					fmt.Printf("Animal %s with ID %d is located in area %s, sector %s", animal.Name, animal.ID, area.Name, sectorName)
+					break finish
+				}
+			}
+		}
+	}
+}
+
+func NewZoo() *Zoo {
+	return &Zoo{}
+}
+
+type Types string
+type Area struct {
+	Name    string
+	Type    Types
+	Sectors map[string]*Sector
+}
+
+func NewArea(n string, t Types) *Area {
+	return &Area{n, t, make(map[string]*Sector)}
+}
+
+type Areas map[string]Area
+type Sector struct {
+	Subtype     string
+	Animals     []*Animal
+	UtilityRoom UtilityRoom
+}
+
+func (s *Sector) AddAnimal(animal *Animal) {
+	s.Animals = append(s.Animals, animal)
+}
+func (s Sector) Describe() {
```

Comment:
І між методами пустий рядок:
```suggestion
}

func (s Sector) Describe() {
```